packages/arb-token-bridge-ui/tests/e2e/specs/batchDeposit.cy.ts

```diff
@@ -0,0 +1,276 @@
+import {
+  ERC20TokenSymbol,
+  getInitialERC20Balance,
+  getInitialETHBalance,
+  getL1NetworkConfig,
+  getL1NetworkName,
+  getL2NetworkConfig,
+  getL2NetworkName,
+  zeroToLessThanOneETH
+} from '../../support/common'
+import { formatAmount } from '../../../src/util/NumberUtils'
+
+describe('Batch Deposit', () => {
+  let parentNativeTokenBalance,
+    parentErc20Balance,
+    childNativeTokenBalance,
+    childErc20Balance: string
+
+  beforeEach(() => {
+    getInitialERC20Balance({
+      tokenAddress: Cypress.env('ERC20_TOKEN_ADDRESS_CHILD_CHAIN'),
+      multiCallerAddress: getL2NetworkConfig().multiCall,
+      address: Cypress.env('ADDRESS'),
+      rpcURL: Cypress.env('ARB_RPC_URL')
+    }).then(val => (childErc20Balance = formatAmount(val)))
+
+    getInitialETHBalance(
+      Cypress.env('ARB_RPC_URL'),
+      Cypress.env('ADDRESS')
+    ).then(val => (childNativeTokenBalance = formatAmount(val)))
+
+    getInitialETHBalance(
+      Cypress.env('ETH_RPC_URL'),
+      Cypress.env('ADDRESS')
+    ).then(val => (parentNativeTokenBalance = formatAmount(val)))
+
+    getInitialERC20Balance({
+      tokenAddress: Cypress.env('ERC20_TOKEN_ADDRESS_PARENT_CHAIN'),
+      multiCallerAddress: getL1NetworkConfig().multiCall,
+      address: Cypress.env('ADDRESS'),
+      rpcURL: Cypress.env('ETH_RPC_URL')
+    }).then(val => (parentErc20Balance = formatAmount(val)))
+  })
+
+  it('should show L1 and L2 chains, and ETH correctly', () => {
+    cy.login({
+      networkType: 'parentChain',
+      url: '/'
+    })
+    cy.findSourceChainButton(getL1NetworkName())
+    cy.findDestinationChainButton(getL2NetworkName())
+    cy.findSelectTokenButton('ETH')
+  })
+
+  it('should deposit erc-20 and native currency to the same address', () => {
+    // randomize the amount to be sure that previous transactions are not checked in e2e
+    const ERC20AmountToSend = Number((Math.random() * 0.001).toFixed(5))
+    const nativeCurrencyAmountToSend = 0.002
+
+    const isOrbitTest = Cypress.env('ORBIT_TEST') == '1'
+    const depositTime = isOrbitTest ? 'Less than a minute' : '9 minutes'
+
+    cy.login({ networkType: 'parentChain' })
+    context('should add a new token', () => {
+      cy.searchAndSelectToken({
+        tokenName: ERC20TokenSymbol,
+        tokenAddress: Cypress.env('ERC20_TOKEN_ADDRESS_PARENT_CHAIN')
+      })
+    })
+
+    context('should show erc-20 parent balance correctly', () => {
+      cy.findByLabelText(`${ERC20TokenSymbol} balance amount on parentChain`)
+        .should('be.visible')
+        .contains(parentErc20Balance)
+        .should('be.visible')
```

Comment:
this is checking one element so there should only be one `.should('be.visible')`, two is redundant

similar for other ones